question_4.py

```diff
@@ -0,0 +1,129 @@
+from datetime import date
+from typing import Any
+
+
+class Restaurant:
+    def __init__(self) -> None:
+        self.__menu = dict()
+        self.__reservations = list()
+        self.__orders = list()
+        self.__GST_RATE = 18
+
+    def add_item_to_menu(self, menu_item: dict[str, Any]) -> None:
+        """
+        This module add new items to menu. menu_item is a dict
+        that must have id, name and price attributes.
+        """
+        if self.__menu.get(menu_item.get("id")) != None:
+            raise Exception("[ERROR] Item already in menu.")
+        else:
+            # pop() will not only return the id but will also it from menu_item
+            self.__menu[menu_item.pop("id")] = menu_item
```

Comment:
Why you pop item from menu?

Reply:
So ... the thing is self.__menu is a dict that is storing menu item like so 
`{ 1: { "name" : "Biryani", "price":220} , 2 :  { "name" : "coke", "price": 120 } }`
where 1 and 2 are item id's respectively. 
However the menu_item that is passed to function as parameter is a dict like 
` { "id": 1, name" : "Biryani", "price":220}`
 So I am removing / pop the id from the menu_item so that I can directly store the rest inside self.__menu